vpr/src/place/noc_place_utils.h

```diff
@@ -0,0 +1,375 @@
+ * re-routes them. The new routes are stored and the NoC cost is updated
+ * to reflect the moved logical router cluster block.
+ * 
+ * @param moved_router_block_id The logical router cluster block that was moved
+ * to a new location during placement.
+ * @param noc_traffic_flows_storage Contains all the traffic flow information
+ * within the NoC. Used to get the traffic flows associated to logical router
+ * blocks.
+ * @param noc_model Contains all the links and rotuters within the NoC. Used
+ * to route traffic flows within the NoC.  
+ * @param noc_flows_router The packet routing algorithm used to route traffic
+ * flows within the NoC.
+ * @param placed_cluster_block_locations A datastructure that identifies the
+ * placed grid locations of all cluster blocks.
+ * @param updated_traffic_flows Keeps track of traffic flows that have been
+ * re-routed. Used to prevent re-rouitng the same traffic flow multiple times.
+ */
+void re_route_associated_traffic_flows(ClusterBlockId moved_router_block_id, NocTrafficFlows& noc_traffic_flows_storage, NocStorage& noc_model, NocRouting* noc_flows_router, const vtr::vector_map<ClusterBlockId, t_block_loc>& placed_cluster_block_locations, std::unordered_set<NocTrafficFlowId>& updated_traffic_flows, int& number_of_affected_traffic_flows);
+
+/**
+ * @brief Used to re-route all the traffic flows associated to logical
+ * router blocks that were supposed to be moved during placement but are
+ * back to their original positions.
+ * 
+ * @param blocks_affected Contains all the blocks that were moved in
+ * the current placement iteration. THis includes the cluster ids of
+ * the moved blocks, their previous locations and their new locations
+ * after being moved.
+ */
+void revert_noc_traffic_flow_routes(const t_pl_blocks_to_be_moved& blocks_affected);
+
+/**
+ * @brief Removes the route of a traffic flow and updates the links to indicate
+ * that the traffic flow does not use them. And then finds
+ * a new route for the traffic flow and updates the links in the new route to
+ * indicate that the traffic flow uses them.
+ * 
+ * @param traffic_flow_id The traffic flow to re-route.
+ * @param noc_traffic_flows_storage Contains all the traffic flow information
+ * within the NoC. Used to get the current traffic flow information.
+ * @param noc_model Contains all the links and rotuters within the NoC. Used
+ * to route traffic flows within the NoC.
+ * @param noc_flows_router The packet routing algorithm used to route traffic
+ * flows within the NoC.
+ * @param placed_cluster_block_locations A datastructure that identifies the
+ * placed grid locations of all cluster blocks.
+ */
+void re_route_traffic_flow(NocTrafficFlowId traffic_flow_id, NocTrafficFlows& noc_traffic_flows_storage, NocStorage& noc_model, NocRouting* noc_flows_router, const vtr::vector_map<ClusterBlockId, t_block_loc>& placed_cluster_block_locations);
+
+/**
+ * @brief 
+ * 
+ * @param new_noc_aggregate_bandwidth_cost 
```

Comment:
Mention this function assumes all traffic flows have been routed.

Reply:
Updated comment